### ISSUE TITLE
Clock doesn't get updated when opening notebook display lid

### ISSUE THREAD

**Issue by xf3** [mate-desktop/mate-panel#481]
On notebook, when closing the lid and then opening it after a while, clock is stuck on date and time the lid was closed at and doesn't get update until clicking it. E.g. when closing notebook in the evening at 23:30 PM and opening it in the morning, instead of showing new date and time, it is stuck at yesterday's date and time. Currently at 1.14.0 version, but issue existed in previous versions.
##

**Comment by lukefromdc** [comment]
That does not sound like a MATE issue but rather an issue with the entire system clock not being updated. Possibly a BIOS/UEFI bug? The vendors only test their code with Windows, meaning Linux kernel devs often hit problems that didn't turn up in Windows testing.  I've got a netbook with battery monitor issues that would require a BIOS update to fix, no matter what the kernel version or DE. I choose to live with it and not bother with a BIOS update on that old 2010 era laptop.

Far worse problems than this, such as not being even able to boot Linux at all without a UEFI update (or worse, a hard to install downgrade) are quite common on newer laptops, to the extent I advise people never to buy new laptops without checking online for known Linux issues first.  Some of this is deliberate interference, as wen Lenovo used UEFI code that looked for the strings "Window 8" or "RHEL" and was set to refuse to boot anything else. You could boot any OS on those by renaming the boot executable, but that identifies Lenovo as a malicious vendor

**Comment by xf3** [comment]
@lukefromdc but the clock is updated once I click the widget to open the calendar, so I guess actually the time value is fine, probably missing some lid event listeners for updating the clock widget? Btw I'm using ASUS N56VZ laptop, BIOS was updated about a year ago.

**Comment by lukefromdc** [comment]
OK, that shows the system time to be OK. I've never had this problem on
my one laptop, but I have always set it up to ignore a closed lid so I can
carry it. I only really use suspend on the desktop, and the clock always
comes back up OK in MATE or any other DE. 

I assume you have it set to suspend when you close the lid? Try suspending
from the poweroff applet/main menu entry and see what happens on resume.
Also if you have any other desktop environments installed try one of them and
see if they also have the same problem.

**Comment by xf3** [comment]
Yes, laptop gets suspended on closing lid. Will try to check whether problem occurs on suspending from power menu. I guess the issue doesn't happen every time, probably only after long suspending period.

**Comment by xf3** [comment]
Tried suspending from power menu - same problem. Btw, time value on lock screen form was correct, so I guess the problem is indeed in clock widget on panel.

**Comment by lukefromdc** [comment]
If clicking on the clock button to open the calendar update the time, that could  mean the applet has the right time but fails to update the label text.  There have been issues before with getting everything to update, usually on a custom panel background change. Still, I have never had this happen, I had enough trouble getting a laptop with a buggy bios NOT to suspend when closing the lid.

To develop a fix someone is going to have to be able to duplicate this-unless someone spots an obvious issue somewhere in the update sequence.

**Comment by xf3** [comment]
Yes, clicking the clock makes it update the time. So what is the next step now, shall I keep the issue opened, probably someone can reproduce it?

**Comment by lukefromdc** [comment]
Keep this open, I really can't do more with it but maybe someone else
can.

**Comment by geki-yaba** [comment]
Well, I have the same issue with Samsung Ultrabook 900X3A from 2011 UEFI mode; without UEFI updates. Let's see if I can find something. Though, this is a low priority issue.

**Comment by geki-yaba** [comment]
Okay, actually it works here. I just have to wait for the next minute to trigger an update showing "date and time without seconds". Showing with seconds it is updated seemingly instantly for the small update interval.

The only improvement I would see, for my case, were to force an update after wakeup if seconds are not shown.

**Comment by kn00tcn** [comment]
suspend as in sleep, not hibernate, right? but does the computer force hibernate once the battery depletes in the suspend mode? (or are we talking about it instantly happening on the next minute)

i dont know how the wake up procedure works at the OS level or at least in regards to the mate-power-manager process, but if it can send an alert to the panel or clock applet, that could be one way

or maybe the panel can trigger something every time its visibility changes or something

i'd want to avoid having some constant check loop every second done by the clock applet though
